black_it/loss_functions/msm.py

```diff
@@ -136,21 +162,33 @@ def compute_loss_1d(
             the MSM loss over a specific coordinate.
         """
         # compute the moments for the simulated ensemble
-        ensemble_real_mom_1d = np.array(
+        ensemble_sim_mom_1d = np.array(
             [self._moment_calculator(s) for s in sim_data_ensemble]
         )
 
         # compute moments of the real time series
-        sim_mom_1d = self._moment_calculator(real_data)
+        real_mom_1d = self._moment_calculator(real_data)
 
-        g = (sim_mom_1d[None, :] - ensemble_real_mom_1d).mean(axis=0)
+        # write moments in relative terms is needed
+        if self._standardise_moments:
+            real_mom_1d = real_mom_1d / abs(real_mom_1d)
+            ensemble_sim_mom_1d = ensemble_sim_mom_1d / (abs(real_mom_1d)[None, :])
 
-        if self._covariance_mat is None:
-            loss_1d = g.dot(g)
-            return loss_1d
+        # mean simulated moments
+        sim_mom_1d = np.mean(ensemble_sim_mom_1d, axis=0)
 
-        W = self._covariance_mat
+        g = real_mom_1d - sim_mom_1d
 
+        if self._covariance_mat == "identity":
+            loss_1d = g.dot(g)
+            return loss_1d
+        if self._covariance_mat == "inverse_variance":
+            W = np.diag(
+                1.0 / np.mean((real_mom_1d[None, :] - ensemble_sim_mom_1d) ** 2, axis=0)
+            )
```

Comment:
```suggestion
        if self._covariance_mat == _CovarianceMatrixType.IDENTITY:
            loss_1d = g.dot(g)
            return loss_1d
        if self._covariance_mat == _CovarianceMatrixType.INVERSE_VARIANCE:
            W = np.diag(
                1.0 / np.mean((real_mom_1d[None, :] - ensemble_sim_mom_1d) ** 2, axis=0)
            )
```

Reply:
BUT this implies that self._covariance_mat is initialized as `_CovarianceMatrixType` rather than `str`.

Alternatively:


```suggestion
        if  _CovarianceMatrixType.IDENTITY(self._covariance_mat) == _CovarianceMatrixType.IDENTITY:
            loss_1d = g.dot(g)
            return loss_1d
        if _CovarianceMatrixType(self._covariance_mat) == _CovarianceMatrixType.INVERSE_VARIANCE:
            W = np.diag(
                1.0 / np.mean((real_mom_1d[None, :] - ensemble_sim_mom_1d) ** 2, axis=0)
            )
```

Reply:
Ok, thanks! I implemented this too, let me know what you think